src/codegen/doc/format.rs

```diff
@@ -57,48 +55,27 @@ fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
     entry
 }
 
-fn format(mut input: &str, env: &Env, in_type: &str) -> String {
+fn format(input: &str, env: &Env, in_type: &str) -> String {
```

Comment:
This was a bad idea actually i guess, it breaks some stuff but most of those are due to it's abuse a bit. It's used in different use cases like being called on `code_blocks_transformation` which doesn't make much sense tbh. The commit that introduced it's usage just says it's improved the formatting of the codeblocks. But then it's also called with `replace_c_types` which means we replace the C code with Rust equivalent? in codeblocks? that doesn't sound correct. 

We should instead just match single  "```" for codeblocks instead of generalizing the function on all the cases.

Reply:
I have fixed this now in the latest commit :)